presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -331,4 +340,52 @@ private Estimate summarizePartitionStatistics(
                             .collect(toImmutableSet()));
         }
     }
+
+    @VisibleForTesting
+    static List<HivePartition> getPartitionsSample(List<HivePartition> partitions, int sampleSize)
+    {
+        checkArgument(sampleSize > 0, "sampleSize is expected to be equal to or greater than zero");
+
+        if (partitions.size() <= sampleSize) {
+            return partitions;
+        }
+
+        List<HivePartition> result = new ArrayList<>();
+
+        int samplesLeft = sampleSize;
+
+        if (samplesLeft > 0) {
+            HivePartition min = partitions.get(0);
+            HivePartition max = partitions.get(0);
+            for (HivePartition partition : partitions) {
+                if (partition.getPartitionId().compareTo(min.getPartitionId()) < 0) {
+                    min = partition;
+                }
+                else if (partition.getPartitionId().compareTo(max.getPartitionId()) > 0) {
+                    max = partition;
+                }
+            }
```

Comment:
I don't like this for loop. It's very efficient, but very verbose as well.
```
Ordering< HivePartition> byPartitionId = ...
HivePartition min = byPartitionId.min(partitions);
HivePartition max = byPartitionId.max(partitions);
```

Reply:
I've done some bench-marking. It is 2.3ms vs 1.6ms =) Iterating over the entire array seems to be expensive. Let's just keep it as is.